common/lib/xmodule/xmodule/js/src/video/08_video_auto_advance_control.js

```diff
@@ -0,0 +1,129 @@
+(function(requirejs, require, define) {
+    'use strict';
+    define(
+        'video/08_video_auto_advance_control.js', [
+            'edx-ui-toolkit/js/utils/html-utils',
+            'underscore'
+        ], function(HtmlUtils, _) {
+            /**
+             * Auto advance control module.
+             * @exports video/08_video_auto_advance_control.js
+             * @constructor
+             * @param {object} state The object containing the state of the video player.
+             * @return {jquery Promise}
+             */
+            var AutoAdvanceControl = function(state) {
+                if (!(this instanceof AutoAdvanceControl)) {
+                    return new AutoAdvanceControl(state);
+                }
+
+                _.bindAll(this, 'onClick', 'destroy', 'autoPlay', 'autoAdvance');
+                this.state = state;
+                this.state.videoAutoAdvanceControl = this;
+                this.initialize();
+
+                return $.Deferred().resolve().promise();
+            };
+
+            AutoAdvanceControl.prototype = {
+                template: HtmlUtils.interpolateHtml(
+                    HtmlUtils.HTML([
+                        '<button class="control auto-advance" aria-disabled="false" title="',
```

Comment:
@mushtaqak [I transformed this to an underscore template](https://github.com/open-craft/edx-platform/commit/e37b7238c4be21150bf70fc98643a69a69a3658f) and it works. The problems are in the tests, they fail I think due to some missing karma config in xmodule which I can't get right. I guessed many paths but I don't know why it doesn't find this one. I didn't find any other test which tests code that uses the „text!“ plug-in.

The error is (note the 404):
```
    time paver test_js -s xmodule
    […]

    Setting up for xmodule javascript
    karma start /edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/js/karma_xmodule.conf.js --single-run=true --capture-timeout=60000 --junitreportpath=/edx/app/edxapp/edx-platform/reports/javascript/javascript_xunit-xmodule.xml --browsers=FirefoxDocker
    ========================================
    Running tests for xmodule javascript
    ========================================
    26 11 2017 22:13:03.141:WARN [watcher]: Pattern "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/js/src/word_cloud/d3.min.js" does not match any file.
    26 11 2017 22:13:03.488:INFO [karma]: Karma v0.13.22 server started at http://edx.devstack.lms:19876/
    changed name to firefox via Selenium Webdriver
    26 11 2017 22:13:03.492:INFO [launcher.selenium-webdriver]: SeleniumWebdriverBrowser (kid:54797300) created
    26 11 2017 22:13:03.493:INFO [launcher]: Starting browser firefox via Selenium Webdriver
    26 11 2017 22:13:03.494:INFO [launcher.selenium-webdriver]: starting firefox via Selenium Webdriver
    26 11 2017 22:13:03.502:INFO [launcher.selenium-webdriver]: sending driver to url http://edx.devstack.lms:19876/?id=54797300
    26 11 2017 22:13:05.559:INFO [Firefox 54.0.0 (Linux 0.0.0)]: Connected on socket W8SNaZUpc1WFijieAAAA with id 54797300
    26 11 2017 22:13:06.303:WARN [web-server]: 404: /base/common/templates/video/autoadvance-button.underscore
    Firefox 54.0.0 (Linux 0.0.0) ERROR
    Error: /edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/js/common/templates/video/autoadvance-button.underscore HTTP status: 404
    at /edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/js/common_static/js/vendor/requirejs/text.js:294
    Firefox 54.0.0 (Linux 0.0.0) ERROR
    Error: /edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/js/common/templates/video/autoadvance-button.underscore HTTP status: 404
    at /edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/js/common_static/js/vendor/requirejs/text.js:294

    Firefox 54.0.0 (Linux 0.0.0): Executed 0 of 0 ERROR (0.771 secs / 0 secs)
```

I declared the .js in the karma config file as seen in latest commits; I also tried to migrate the test file to RequireJS and fix a small bug (as explained [here](https://github.com/edx/edx-platform/pull/15803#discussion_r153159088)).
Would you have any direction or model to follow to make the xmodule tests run? If this seems more unneedingly complex, could we get back to the HTML inside the JS? Thanks

Reply:
Sorry @mushtaqak, we'd really like to use your suggested template approach, but we're way over budget on this change, and fixing up the tests is just taking too long!

So I've had to submit a62d885 to revert back to using the inline HTML instead of the underscore template, like the [other video JS does](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/js/src/video/09_video_caption.js#L81-L98).  Hopefully edX or someone will figure out a way to use fragments for all the video XBlock html pieces, and then it'll all be cleaner.

See these commits for the attempted fixes:  b79dbaa...78a75df, 6b7f085